sway/desktop/layer_shell.c

```diff
@@ -234,7 +235,9 @@ void arrange_layers(struct sway_output *output) {
 		if (topmost != NULL) {
 			seat_set_focus_layer(seat, topmost->layer_surface);
 		} else if (seat->focused_layer &&
-				!seat->focused_layer->current.keyboard_interactive) {
+				seat->focused_layer->current.keyboard_interactive
+					!= ZWLR_LAYER_SURFACE_V1_KEYBOARD_INTERACTIVITY_EXCLUSIVE) {
+			seat->exclusive_layer = NULL;
```

Comment:
Note that there can be  multiple layer surfaces asking for exclusive keyboard interactivity, with the topmost one getting the privilege.

Maybe a simple "update_exclusive_layer" that iterates through them and picks the first one?

Reply:
Don't the lines above do the same thing? Or am I misunderstanding what you've meant? 😅

Reply:
D'oh, misread. Testing with two exclusive interactivity layers would be a good test though.

Reply:
> D'oh, misread. Testing with two exclusive interactivity layers would be a good test though.

Works fine, the layer focus remains until I move the cursor/click inside of the other exclusive layer, and vise versa :)